config/v1/types_infrastructure.go

```diff
@@ -1647,8 +1665,12 @@ type IBMCloudPlatformStatus struct {
 	DNSInstanceCRN string `json:"dnsInstanceCRN,omitempty"`
 
 	// serviceEndpoints is a list of custom endpoints which will override the default
-	// service endpoints of an IBM Cloud service. These endpoints are consumed by
-	// components within the cluster to reach the respective IBM Cloud Services.
+	// service endpoints of an IBM service. These endpoints are used by components
+	// within the cluster when trying to reach the IBM Cloud Services that have been
+	// overriden. The CCCMO reads in the IBMCloudPlatformSpec and validates each
+	// endpoint is resolvable. Once validated, the cloud config and IBMCloudPlatformStatus
+	// are updated to reflect the same custom endpoints.
+	// +kubebuilder:validation:MaxItems=13
```

Comment:
There is a feature gate aware maxItems, that we should use here
```suggestion
	// +openshift:validation:FeatureGateAwareMaxItems:featureGate=DyanmicServiceEndpointIBMCloud,maxItems=13
```